client/src/components/TransactionItemAdder/TransactionItemAdder.tsx

```diff
@@ -83,10 +86,15 @@ const TransactionItemAdder: React.FC<ITransactionItemAdder> = ({
             : +currentAmount,
       }));
     await refetchBudget();
-    reset();
+    reset({
+      title: "",
+      amount: "",
+    });
+    setTimeout(() => {
```

Comment:
What is this setTimeout for?

Reply:
Its to remove the error alert after a user submits the form. Theres an issue of show an error alert once user submits because react hook form sets the form back to an empty string, but yup validator is still validating it and spitting out an error saying "field is required". So I implemented the settimeout to run 100ms after submitting so the error disappears until triggered again by user error

Reply:
I tried just using the clearError without the setTimeout function, but I think it does it simultaneously so the error alert is still popping up. With the 100ms delay, I think it reliably clears all errors after the fields have been reset

Reply:
Hmm, that can work for now.  But it's not very scalable.  We can keep it for now, but just know that it's not really the best solution.

Reply:
Are setTimeout functions expensive to run?